pkg/controller/deployment/deployment_controller.go

```diff
@@ -487,13 +493,13 @@ func (dc *DeploymentController) handleErr(err error, key interface{}) {
 	}
 
 	if dc.queue.NumRequeues(key) < maxRetries {
-		klog.V(2).InfoS("Error syncing deployment", "deployment", klog.KRef(ns, name), "err", err)
+		klog.V(2).InfoS("Error syncing deployment", klogKeyDeployment, klog.KRef(ns, name), "key", key, "err", err)
```

Comment:
why introduce "key" key here? Based on name and log message it's impossible to interpret what value is stored under it without looking at the source code. I would omit  this key as this log message is not related to cache key at all.

Reply:
when `cache.SplitMetaNamespaceKey` in [line 484](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/deployment_controller.go#L484)  returns an error, both `ns` and `name` are empty strings.
The message we get here in such situation will looks like that: Error syncing deployment deployment=""
`key` was already introduced in line 486, and I reused it to provide more information when such problem occurs.

Another options I can think of is to use `key.(string)` as deployment name instead of klog.KRef(ns, name) if `keyErr` != nil.
Let me know please if you prefer to go this direction.

Reply:
removed `key`.